src/vs/workbench/contrib/terminal/browser/xterm/suggestAddon.ts

```diff
@@ -357,22 +370,20 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 
 			// Send the completion
 			this._onAcceptedCompletion.fire([
-				// TODO: Right arrow to end of the replacement
-				// Left arrow to end of the replacement
-				'\x1b[D'.repeat(Math.max(suggestion.model.replacementLength - this._cursorIndexStart + this._cursorIndexDelta, 0)),
-				// Delete to remove additional input
-				'\x1b[3~'.repeat(this._additionalInput?.length ?? 0),
+				// Right arrow to the end of the additional input
+				'\x1b[C'.repeat(Math.max((this._additionalInput?.length ?? 0) - this._cursorIndexDelta, 0)),
+				// Backspace to remove additional input
+				'\x7F'.repeat(this._additionalInput?.length ?? 0),
 				// Backspace to remove the replacement
 				'\x7F'.repeat(suggestion.model.replacementLength),
 				// Write the completion
 				suggestion.item.completion.label,
 			].join(''));
 
 			// Disable completions triggering the widget temporarily to avoid completion requests
-			// caused by the completion itself to show.
-			this._enableWidget = false;
-			// TODO: Disable the widget in a more sophisticated way
-			timeout(100).then(e => this._enableWidget = true);
+			// caused by the completion itself to show
+			this._cacheCurrentCompletion();
+			timeout(300).then(e => this._cachedCompletions = new Set());
```

Comment:
Generally timeouts are inherently flaky, I forgot why this was needed but is there some other event we could use to clear this state?

Reply:
Removed the timeouts since we won't receive suggestions during the accepted completion